### PR TITLE
feat: add option `omit_force_resolver_fields`

### PR DESCRIPTION
**Pull request by abekoh**
This PR introduces a new option, `omit_force_resolver_fields` . When enabled, fields annotated with `@goField(forceResolver: true)` will not be generated by the modelgen plugin.

I frequently use the `forceResolver` option. However, fields with this option get generated in the same manner as those without it, leading to instances where I have mistakenly implemented mapping. This change is intended to prevent such mistakes.

I would greatly appreciate any suggestions or improvements.

Relevant issue: #1536

I have:
 - [x] Added tests covering the bug / feature (see [testing](https://github.com/99designs/gqlgen/blob/master/TESTING.md))
 - [x] Updated any relevant documentation (see [docs](https://github.com/99designs/gqlgen/tree/master/docs/content))

### CODE REVIEW

File: docs/content/config.md [hunk 1]
```diff
@@ -71,6 +71,9 @@ resolver:
 # Optional: wrap nullable input fields with Omittable
 # nullable_input_omittable: true
 
+# Optional: omit generated model fields with `@goField(forceResolver: true)`
+# omit_force_resolver_fields: true
```
Comment (chadxzs) [inline docs/content/config.md:75]: As someone relatively new to gqlgen, I don't think I would know what this option does without looking at the source code if I only had this documentation as a guide. Is there some way to make the comment here more illustrative, or some way you can provide additional context on when someone may want to do this?

Comment (abekoh) [reply docs/content/config.md:77]: Thanks for your reviewing! I fixed at eb6ed62cf3a86229b7289f25a8456571364bd3b9

**Comment by chadxzs** [comment]
@abekoh FYI I'm not a maintainer here, simply reviewing PRs as a way to help out some and become more familiar with the codebase. @StevenACoffman is primarily maintaining the codebase right now so you'll need to wait until he can review it.

**Comment by UnAfraid** [comment]
I would prefer a bit more granular control over that feature within @goField(omitForceResolverField: true) directive also for cases where omitting is not an option.

Consider the following example: 
The Post wouldn't necessarily have all the user data, but only the user id, so i would initialise new User object only with the id, then use data loader to load the user by the id.

```graphql
type User {
  id: ID!
  email: String!
  posts: [Post!]! @goField(forceResolver: true) # This field can be omitted
}

type Post {
  id: ID!
  title: String!
  body: String!
  createdBy: User! @goField(forceResolver: true) # Here user id would be needed to know which user to load
  updatedBy: User  @goField(forceResolver: true) # ^
  deletedBy: User  @goField(forceResolver: true) # ^
  createdAt: DateTime!
  #...
}
```

**Comment by abekoh** [comment]
@UnAfraid 
Thank you for your feedback! Indeed, in such cases, we would not want to omit it. I will consider a different option.

**Comment by mlaflamm** [comment]
I know this PR is old and closed but for those ending here, you can rename the go field to start with a lowercase letter to unexport it. This is almost equivalent to omitting it.

You can also replace a struct field exported via a resolver by changing its name and its type.

Here is the adapted example previously provided by @UnAfraid:

```graphql
type User {
  id: ID!
  email: String!
  posts: [Post!]! @goField(forceResolver:true, name: "posts") # This field is unexported
}

type Post {
  id: ID!
  title: String!
  body: String!
  createdBy: User! @goField(forceResolver: true, name: "CreatedByUserID", type: "string") @goTag(key: "json", value: "-")
  updatedBy: User  @goField(forceResolver: true, name: "UpdatedByUserID", type: "*string") @goTag(key: "json", value: "-")
  deletedBy: User  @goField(forceResolver: true, name: "DeletedByUserID", type: "*string") @goTag(key: "json", value: "-")
  createdAt: DateTime!
  #...
}
```

**Pull request closed without merging**